pkg/crc/api/api_http.go

```diff
@@ -1,215 +1,66 @@
-		stopResult := handler.Stop()
-		sendResponse(w, stopResult)
-	})
-
-	mux.HandleFunc("/poweroff", func(w http.ResponseWriter, r *http.Request) {
-		if wrongHTTPMethodUsed(r, w, http.MethodPost) {
-			return
-		}
-		stopResult := handler.PowerOff()
-		sendResponse(w, stopResult)
-	})
+	server := newServer()
 
-	mux.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) {
-		if wrongHTTPMethodUsed(r, w, http.MethodGet) {
-			return
-		}
-		status := handler.Status()
-		sendResponse(w, status)
-	})
+	server.POST("/start", handler.Start)
+	server.GET("/start", handler.Start)
 
-	mux.HandleFunc("/delete", func(w http.ResponseWriter, r *http.Request) {
-		if wrongHTTPMethodUsed(r, w, http.MethodGet, http.MethodDelete) {
-			return
-		}
-		deleteResult := handler.Delete()
-		sendResponse(w, deleteResult)
-	})
+	server.POST("/stop", handler.Stop)
+	server.GET("/stop", handler.Stop)
 
-	mux.HandleFunc("/version", func(w http.ResponseWriter, r *http.Request) {
-		if wrongHTTPMethodUsed(r, w, http.MethodGet) {
-			return
-		}
-		versionResult := handler.GetVersion()
-		sendResponse(w, versionResult)
-	})
+	server.POST("/poweroff", handler.PowerOff)
+	server.GET("/status", handler.Status)
 
-	mux.HandleFunc("/webconsoleurl", func(w http.ResponseWriter, r *http.Request) {
-		if wrongHTTPMethodUsed(r, w, http.MethodGet) {
-			return
-		}
-		webconsoleInfo := handler.GetWebconsoleInfo()
-		sendResponse(w, webconsoleInfo)
-	})
+	server.DELETE("/delete", handler.Delete)
+	server.GET("/delete", handler.Delete)
 
-	mux.Handle("/config/", http.StripPrefix("/config", newConfigMux(handler)))
+	server.GET("/version", handler.GetVersion)
 
-	mux.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
-		sendResponse(w, handler.Logs())
-	})
-
-	mux.HandleFunc("/telemetry", func(w http.ResponseWriter, r *http.Request) {
-		data, err := verifyRequestAndReadBody(w, r, http.MethodGet, http.MethodPost)
-		if err != nil {
-			return
-		}
-		sendResponse(w, handler.UploadTelemetry(data))
-	})
-
-	mux.HandleFunc("/pull-secret", pullSecretHandler(config))
-
-	return mux
-}
-
-func pullSecretHandler(config crcConfig.Storage) func(w http.ResponseWriter, r *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		if r.Method == http.MethodPost {
-			bin, err := ioutil.ReadAll(r.Body)
-			if err != nil {
-				http.Error(w, err.Error(), http.StatusBadRequest)
-				return
-			}
-			if err := cluster.StoreInKeyring(string(bin)); err != nil {
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
-			}
-			w.WriteHeader(http.StatusCreated)
-			return
-		}
-		if _, err := cluster.NewNonInteractivePullSecretLoader(config, "").Value(); err == nil {
-			w.WriteHeader(http.StatusOK)
-		} else {
-			w.WriteHeader(http.StatusNotFound)
-		}
-	}
-}
+	server.GET("/webconsoleurl", handler.GetWebconsoleInfo)
 
-func newConfigMux(handler *Handler) http.Handler {
-	mux := http.NewServeMux()
+	server.GET("/config/set", handler.SetConfig)
```

Comment:
hopefully this makes it easier to refactor these; as these do not follow the verbs #2580